go/libraries/doltcore/doltdb/root_val.go

```diff
@@ -291,6 +295,30 @@ func (root *rootValue) HandlePostMerge(ctx context.Context, ourRoot, theirRoot,
 	return root, nil
 }
 
+func (root *rootValue) GetTableSchemaHash(ctx context.Context, tName TableName, overridden bool) (hash.Hash, error) {
```

Comment:
Get rid of overridden here. The caching should always happen when you call this method. If they don't want to cache the new result, don't call this method.

Reply:
I'm back and forth on whether the problem is the way overridden is implemented...like it's easy to move the switch to all points where we read the schema hash, but then the behavior is implicit instead of explicit.